snapcraft_legacy/internal/project_loader/_extensions/ros1_noetic.py

```diff
@@ -78,6 +78,7 @@ def __init__(self, *, extension_name: str, yaml_data: Dict[str, Any]) -> None:
                 {"ROS_VERSION": self.ROS_VERSION},
                 {"ROS_DISTRO": self.ROS_DISTRO},
             ],
+            "stage-packages": [f"ros-{self.ROS_DISTRO}-ros-environment"]
```

Comment:
Why do you need to stage this package?
It will be added even to snaps that are not leveraging the content-sharing feature.
Shouldn't this be added in the final snap when using content-sharing ros snap feature?

Reply:
This package installs, among other things, the bash scripts that set up the ROS environment when sourcing a workspace (e.g. `ROS_PACKAGE_PATH`). If these scripts aren't present when sourcing, some ROS env var aren't set and thus ROS packages can't be found etc. Note that this package is virtually always staged being near the root of any ROS dependency tree. It is only in the content-sharing case that this package may not be staged (since it is found in the content-sharing snap) and thus results in half-broken ROS workspace.